src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -115,13 +135,23 @@ def backfill_seer_grouping_records(
 
     last_group_id = groups_to_backfill_with_no_embedding[-1]
 
-    snuba_results = get_data_from_snuba(project, groups_to_backfill_with_no_embedding)
+    # Run single snuba query per group for high event counts, otherwise run bulk query
+    snuba_results_high_count = get_data_from_snuba_single_group_query(
+        project, group_id_last_seen_no_embeddings_high_count
+    )
+    snuba_results_low_count = get_data_from_snuba_bulk_groups_query(
+        project, group_id_last_seen_no_embeddings_low_count
+    )
```

Comment:
I think it would be cleaner if you put this logic into one function which this main function would call, and then you'd combine the results in that function. so that way further down in the logic, it doesn't have to care what a high result / low result is, it's all snuba results that are treated the same